cmd/internal/configvalidator/validate.go

```diff
@@ -0,0 +1,102 @@
+package configvalidator
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// ErrUnknowField returns when an unknown field appears in the config.
+var ErrUnknowField = errors.New("unknown field")
+
+// CheckForUnknownFields validate the config struct for unknown fields with the config map.
+// If the field in the config is a map of a key and some value, need to use `mapstructure:,remain` and
+// `prefix` tag with the key prefix, even it is empty string.
+func CheckForUnknownFields(configMap map[string]any, config any, currentPath string) error {
+	expectedFields := getFieldsFromStruct(config)
+
+	for key, val := range configMap {
```

Comment:
Seems to be _very_ inefficient since you traverse the same items for every key. Maybe it's not a problem given the size of our configuration files, but I'd try flattening (into a slice) and sorting `configMap` which allows to walk by prefixes more effectively then. Still, what's the time we spend here for example config?

Reply:
For `ir` example config, the execution time of this function is 192 microseconds, for `node` - 197 microseconds. 
It's going to be a little difficult considering `,remain` case and not so necessary now.